### DISCUSSION TITLE
cartCreate mutation lines input is useless and cartCreate is not intrusive

### DISCUSSION THREAD

**Discussion by jullite** [Shopify/storefront-api-feedback#165] [Help] [answered]
Hi all!

The cartCreate mutation only returns the cart id, and the lines of products didn't add to cart. Why we need this input?

this is my mutation

![image](https://user-images.githubusercontent.com/86504730/211277450-d775528f-76ca-4d96-8edc-2b15805ea341.png)

it returns

![image](https://user-images.githubusercontent.com/86504730/211277472-c26cd484-5911-44c4-8eb4-cd8df55fc923.png)

maybe there is two options:
1. remove the lines input
2. remove the cartCreate mutation, create when we use lines-add https://github.com/Shopify/storefront-api-feedback/discussions/12#discussioncomment-1803927

**Comment by vixdug** [comment, marked as answer]
Hey @jullite 

with GraphQL you can request the fields that you want to be returned. We have an example found here; https://shopify.dev/docs/custom-storefronts/building-with-the-storefront-api/cart/manage#step-1-create-a-cart-and-add-a-line-item

**Reply by jullite** [reply]
Thank you so much for your response!
I have figured out the puzzle... It's my fault. I forgot to pass variables on my mutation. So sorry to bother you.
